src/main/java/org/jenkins/tools/test/model/PluginCompatTesterConfig.java

```diff
@@ -399,7 +398,12 @@ public String getTestJavaVersion() throws IOException {
             javaCmdAbsolutePath = "java";
         }
         final Process process = new ProcessBuilder().command(javaCmdAbsolutePath, "-XshowSettings:properties -version").redirectErrorStream(true).start();
-        final String javaVersionOutput = IOUtils.toString(process.getInputStream());
+        try {
+            process.waitFor();
+        } catch (InterruptedException e) {
+            throw new IOException("interrupted while getting Java version", e);
+        }
```

Comment:
Also the two arguments were previously being passed as one argument, which failed on Windows once the `waitFor` was (correctly!) introduced. Separating them as two Java arguments to the `ProcessBuilder` API resolves the issue.

Reply:
> we were not waiting for the process to terminate before trying to read its output.

for good reason see above ;-) 

`readAllBytes` will block until the end of stream is reached - (ie the process has terminated, or otherwise closed its output stream)  thus we can not have partial data.

the argument splitting fix is still valid.